### DISCUSSION TITLE
Better searchbar/function on http://mathjs.org website using (e.g.) Algolia

### DISCUSSION THREAD

**Discussion by JasonShin** [josdejong/mathjs#2756] [Ideas]
Hello,

This is slightly subjective but can we implement a different search functionality on the website? I found http://mathjs.org/docs/index.html was confusing to navigate initially. I've been using (also contributing to) math.js so I'm quite familiar with the structure of the library now, but I'd still appreciate a better documentation navigation.

Few findings of the current _Google Custom Search_
1. Search bar does not show up when you have adblocker or Ghostery enabled. I am using Ghostery at the moment and it prevents the search bar from showing up
2. When it fails to find any search hits from http://mathjs.org, then it shows links to other websites

Why Algolia?
I'm not advocating Algolia here but mentioned it as an example. If you decide to use it, you can apply for an open source project license to use their platform for [free](https://www.algolia.com/for-open-source).

I welcome any replies and thoughts.

**Comment by JasonShin** [comment]
If we agree to settle on a solution, I am happy to help out in the process of analysis and implementing.

**Comment by josdejong** [comment]
Great idea, and thanks for your offer to put effort into improving search on the website @JasonShin. I love the search of Algolia, have seen it on websites like github itself and vuejs. We can also look at how sites like `lodash` and `date-fns` have implemented search, works nicely on these sites too. I don't have a strong opinion here though.

The website of mathjs is currently a static, markdown generated website. Would be easiest if the search solution we choose supports that out of the box.

**Comment by harrysarson** [comment]
Lodash's docs are pretty good in my opinion. Aiming for something similar for mathjs would not be a bad shout

**Reply by dvd101x** [reply]
I agree about [lodash documentation](https://lodash.com/docs/4.17.15), I find the navbar very usefull and would love to have that quick access to functions documentation in mathjs.

I recently found a library that might help and it includes the search.

https://docsify.js.org

**Reply by josdejong** [reply]
Yeah, something like that would be great. Anyone able to work out a better search solution on the website? Help would be welcome.

**Reply by dvd101x** [reply]
Hi, I just did a quick test with docsify including the search plugin and it seems kind of nice.
<img width="1065" alt="docsify test" src="https://github.com/josdejong/mathjs/assets/16261897/b97f4162-9a08-447c-9a10-8ff5303c8165">

This is with the docs as they are, but it's not finding functions and other stuff.

**Reply by dvd101x** [reply]
Might also include latex with

https://scruel.github.io/docsify-latex/#/

**Reply by josdejong** [reply]
Ow, yes it would be really useful to be able to use LateX in the docs.
